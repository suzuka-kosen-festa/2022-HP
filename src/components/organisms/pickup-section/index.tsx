import type { FC } from "react";
import tw from "twin.macro";
import { Link } from "wouter";
import type { PickupSectionProperties } from "./types/model";
import { Heading } from "../../atoms/heading";
import { PickupCard } from "../../molecules/pickup-card";
import { Button } from "../../atoms/button";

const PickupSectionContainer = tw.section`flex flex-col justify-center items-center`;

const CardList = tw.ul`grid mb-6 grid-cols-1 gap-2 sm:(mb-16 grid-cols-2 gap-6)`;

const PickupSection: FC<PickupSectionProperties> = ({ title, buttonText, events, ...rest }) => (
  <PickupSectionContainer aria-label="pick up event" {...rest}>
    <Heading css={tw`mb-4 sm:mb-16`} colorKey="green">
      {title}
    </Heading>
    <CardList>
      {events.map(event => (
        <PickupCard event={event} key={event.title} />
      ))}
    </CardList>
    <Link to="/timetable?select=30">
      <Button colorKey="green" as="a" role="button">
        {buttonText}
      </Button>
    </Link>
  </PickupSectionContainer>
);

export { PickupSection };
