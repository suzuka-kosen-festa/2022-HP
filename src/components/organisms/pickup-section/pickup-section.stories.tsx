import type { ComponentStoryObj, ComponentMeta } from "@storybook/react";
import { ComponentPropsWithoutRef } from "react";
import { PickupSection } from ".";

type T = typeof PickupSection;
type Story = ComponentStoryObj<T>;
type Meta = ComponentMeta<T>;

const args: ComponentPropsWithoutRef<T> = {
  title: "Pick up events",
  cardData: [
    {
      title: "へぇ王",
      description: "先生たちが自分の研究を紹介！4年生は卒研部を考えるいい機会に！高専生、中学生に向けた勉強イベント！",
      startTime: "2022-10-30 11:30",
      stage: "Main",
    },
    {
      title: "TTTGP",
      description: "その名もTop of TikTok GP！TikTokで流行ったダンス、音源、TikTokerなどTikTokに関するクイズを出題！",
      startTime: "2022-10-30 13:15",
      stage: "Main",
    },
    {
      title: "俺1GP",
      description: "鈴鹿高専のお笑いライブ！笑いで全てを包み込め！",
      startTime: "2022-10-31 13:00",
      stage: "Main",
    },
    {
      title: "メモリーズ",
      description: "5年生のダンス披露！みんなで頑張って踊ります！",
      startTime: "2022-10-31 14:15",
      stage: "Main",
    },
  ],
  buttonText: "タイムスケジュールを見る",
};

export default {
  args,
  argTypes: {
    buttonText: {
      description: "The text for button in section",
      control: {
        type: "text",
      },
    },
    cardData: {
      description: "Data for pick up event card",
      control: {
        type: "array",
      },
    },
    title: {
      description: "Title of section",
      control: {
        type: "text",
      },
    },
  },
  component: PickupSection,
} as Meta;

export const Default: Story = {};
