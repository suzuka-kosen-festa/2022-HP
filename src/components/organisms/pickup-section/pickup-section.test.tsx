import { composeStories } from "@storybook/testing-react";
import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import * as stories from "./pickup-section.stories";

const { Default } = composeStories(stories);

const options = {
  name: "pick up event",
};

describe("(components) organisms/pickup-section", () => {
  test("to be organisms", () => {
    const { container } = render(<Default />);
    expect(container).toBeOrganism();
  });
  test("to be [role=region]", () => {
    const { getByRole } = render(<Default />);
    expect(getByRole("region", options));
  });
  test("take snap shot", () => {
    const { container } = render(<Default />);
    expect(container).toMatchSnapshot();
  });
});
