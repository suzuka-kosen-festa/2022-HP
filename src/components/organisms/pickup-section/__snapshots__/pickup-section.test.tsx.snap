// Vitest Snapshot v1

exports[`(components) organisms/pickup-section > take snap shot 1`] = `
<div>
  <section
    aria-label="pick up event"
    class="css-wkpw2c"
  >
    <h1
      class="css-16hzca7-PickupSection"
    >
      Pick up events
    </h1>
    <ul
      class="css-9y1ynn"
    >
      <li
        class="css-10pxw4f"
      >
        <h2
          class="css-1a2c2qr"
        >
          へぇ王
        </h2>
        <p
          class="css-y87pn1"
        >
          先生たちが自分の研究を紹介！4年生は卒研部を考えるいい機会に！高専生、中学生に向けた勉強イベント！
        </p>
        <div
          class="css-18lwsxv"
        >
          <p
            class="css-y87pn1"
          >
            日時: 
            10/30 (日) 11:30~
          </p>
          <p
            class="css-y87pn1"
          >
            場所: 
            メインステージ
          </p>
        </div>
      </li>
      <li
        class="css-10pxw4f"
      >
        <h2
          class="css-1a2c2qr"
        >
          TTTGP
        </h2>
        <p
          class="css-y87pn1"
        >
          その名もTop of TikTok GP！TikTokで流行ったダンス、音源、TikTokerなどTikTokに関するクイズを出題！
        </p>
        <div
          class="css-18lwsxv"
        >
          <p
            class="css-y87pn1"
          >
            日時: 
            10/30 (日) 13:15~
          </p>
          <p
            class="css-y87pn1"
          >
            場所: 
            メインステージ
          </p>
        </div>
      </li>
      <li
        class="css-10pxw4f"
      >
        <h2
          class="css-1a2c2qr"
        >
          俺1GP
        </h2>
        <p
          class="css-y87pn1"
        >
          鈴鹿高専のお笑いライブ！笑いで全てを包み込め！
        </p>
        <div
          class="css-18lwsxv"
        >
          <p
            class="css-y87pn1"
          >
            日時: 
            10/31 (月) 13:00~
          </p>
          <p
            class="css-y87pn1"
          >
            場所: 
            メインステージ
          </p>
        </div>
      </li>
      <li
        class="css-10pxw4f"
      >
        <h2
          class="css-1a2c2qr"
        >
          メモリーズ
        </h2>
        <p
          class="css-y87pn1"
        >
          5年生のダンス披露！みんなで頑張って踊ります！
        </p>
        <div
          class="css-18lwsxv"
        >
          <p
            class="css-y87pn1"
          >
            日時: 
            10/31 (月) 14:15~
          </p>
          <p
            class="css-y87pn1"
          >
            場所: 
            メインステージ
          </p>
        </div>
      </li>
    </ul>
    <a
      class="css-3os8bq-Button"
      href="/timetable"
      role="button"
      tabindex="0"
    >
      <p
        class="css-7bzdgk-Button"
      >
        タイムスケジュールを見る
      </p>
    </a>
  </section>
</div>
`;
